src/pubsub/ua_pubsub_writer.c

```diff
@@ -690,6 +690,9 @@ UA_DataSetWriter_generateDataSetMessage(UA_Server *server,
             dsm->configuredSize = 0;
         }
 
+        /* setting configured size in the dataSetMessage to add padding later on */
+        dataSetMessage->configuredSize = dsm->configuredSize;
```

Comment:
Where does the value come from? UA_UadpDataSetWriterMessageDataType ?

Reply:
Yes. See the added testcase.